content/graphql/guards-interceptors.md

```diff
@@ -85,3 +85,27 @@ async upvotePost(
 ```
 
 > info **Hint** In the above example, we have assumed that the `user` object is assigned to the context of your GraphQL application.
+
+#### Interceptors at the `@ResoleField()` method level
+
+In a GraphQL context, interceptors [do not access the full GraphQL response](https://github.com/nestjs/graphql/issues/320#issuecomment-511193229): they only get the response of the `@Query()`/`@Mutation()` method, but not the complete response with the resolved fields. You can tell Nest to execute interceptors (as well as guards and filters) for methods annotated `@ResolveField()` by setting the `fieldResolverEnhancers` option in `GqlModuleOptions`:
+
+```typescript
+GraphQLModule.forRoot({
+  fieldResolverEnhancers: ['interceptors']
+}),
+```
+
+> **Warning** Enabling interceptors for field resolvers can cause performance issues when you are returning lots of records and your field resolver is executed thousands of times. For this reason, when you enable `fieldResolverEnhancers`, we advise you to skip execution of interceptors that are not strictly necessary for your field resolvers. You can do this using the following helper method:
+
+```typescript
+export function isResolvingGraphQLField(context: ExecutionContext): boolean {
+  if (context.getType().toString() === 'graphql') {
```

Comment:
One little suggestion: you can use the `GqlContextType` as stated here https://docs.nestjs.com/fundamentals/execution-context#current-application-context

```typescript
if (context.getType<GqlContextType>() === 'graphql') {
```